From ab389a24ee9c31fc22cbc9eab1ef1db4f53efb7b Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Tue, 22 Feb 2022 05:36:59 +0200
Subject: [PATCH] [SW-72680] greco: pci: enable hot rst control by fw

So far FW is managing the response to pci hot reset, but the HW
doesn't know about it, so both FW and HW are reacting to the hot
reset and device is going to bad state.
In order to avoid that, the relevant register is updated flagging
the HW that FW takes control of the hot reset response.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Idbc55ac747a8c2eaa4664339863584c08e5d0fbb
---

diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index 5516a59..0fc49de 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -251,6 +251,12 @@
 	return mmCPU_TIMESTAMP_CNTCR;
 }
 
+void pci_dev_specific_setup(void)
+{
+	/* FW takes responsibility to process HOT reset */
+	regmap_writel(1, mmPCIE_WRAP_CPU_HOT_RST);
+}
+
 void pci_dev_specific_hot_rst(void)
 {
 	regmap_writel(0x10, mmPCIE_CORE_HOT_RST);
